packages/nvidia_nat_nemo_customizer/src/nat/plugins/customizer/dpo/trainer_adapter.py

```
@@ -0,0 +1,550 @@
+    async def _ensure_namespaces_exist(self) -> None:
+        """Create namespaces in entity store and datastore if they don't exist."""
+        namespace = self.adapter_config.namespace
+
+        # Create namespace in entity store
+        try:
+            self.entity_client.namespaces.create(
+                id=namespace,
+                description=f"NAT finetuning namespace: {namespace}",
+            )
+            logger.info(f"Created namespace '{namespace}' in Entity Store")
+        except Exception as e:
```

Ruff (0.14.8)

Do not catch blind exception: `Exception`

(BLE001)